File: server/schemas/report.py
from pydantic import BaseModel, Field


class ExtractedTest(BaseModel):
    name: str
    value: str
    unit: str | None = None
    referenceRange: str | None = None
    status: str = Field(default="unknown", pattern="^(low|normal|high|unknown)$")
    explanation: str


class ReportSummary(BaseModel):
    id: str
    patientName: str | None = None
    reportDate: str | None = None
    language: str
    tests: list[ExtractedTest]
    summary: str
    lifestyleTips: list[str]
    disclaimer: str


class ChatMessage(BaseModel):
    role: str = Field(pattern="^(user|assistant)$")
    content: str


class ChatRequest(BaseModel):
    messages: list[ChatMessage]

File: client/src/types/report.ts
export type TestStatus = "low" | "normal" | "high" | "unknown";

export type ExtractedTest = {
  name: string;
  value: string;
  unit?: string;
  referenceRange?: string;
  status: TestStatus;
  explanation: string;
};

export type ReportSummary = {
  id: string;
  patientName?: string;
  reportDate?: string;
  language: string;
  tests: ExtractedTest[];
  summary: string;
  lifestyleTips: string[];
  disclaimer: string;
};

export type ChatMessage = {
  role: "user" | "assistant";
  content: string;
};

